src/pyggdrasil/tree_inference/_file_id.py

```diff
@@ -156,6 +172,52 @@ def _create_id(self) -> str:
     def __str__(self) -> str:
         return self.id
 
+    @classmethod
+    def from_str(cls, str_id: str):
+        """Creates a CellSimulation id from a string representation of the id.
+        Args:
+            str_id: str
+
+        Throws:
+            AssertionError if the string representation is not valid
+        """
+        # split string by underscore and assign to attributes
+        # CS_1-T_d_10-100_0.01_0.01_0.01_true_UXR
+        cs_par1, tree_id, cs_part2 = str_id.split("-")
+        # check prefix and postfix
+        assert cs_par1.startswith("CS_")
+        assert tree_id.startswith("T_")
+        assert cs_part2.endswith("UXR") or cs_part2.endswith("UIR")
+        # split cs_part2 by underscore and assign to attributes
+        cs_part2 = cs_part2.split("_")
```

Comment:
This logic is very complex – at least a little amount of unit tests acting both as sanity checks and examples what to expect would be helpful in the long run.

Reply:
Added one - caught mistakes actually. Good call !